server/graphql/v1/mutations/tiers.js

```diff
@@ -6,6 +7,15 @@ export function editTiers(_, args, req) {
     throw new errors.Unauthorized('You need to be logged in to edit tiers');
   }
 
+  const tiers = map(args.tiers, tier => {
+    if (tier.presets && !tier.minimumAmount) {
```

Comment:
I find there are some ambiguity in the current code, the following one would be clearer:

```
if (!tier.minimumAmount) {
  if (tier.presets && tier.presets.length > 0 && tier.amount) {
     tier.minimumAmount = Math.min(tier.amount, ...tier.presets);
  } else if (tier.presets && tier.presets.length > 0) {
    tier.minimumAmount = Math.min(...tier.presets);
  } else if (tier.amount) {
    tier.minimumAmount = tier.amount;
  }
}
 ```

What do you think?

Reply:
Also make sure the logic exactly match the one in the migration for consistency.